### PR TITLE
Python format() built-in-function entry

### LINKED ISSUE #2525: [Term Entry] Python Built-in Functions format()
### Reason/inspiration (optional)

We would like a new term entry on the format() Python: [Built-in Function](https://www.codecademy.com/resources/docs/python/built-in-functions). The entry should go in a new file under `docs/content/python/concepts/built-in-functions/terms/format/format.md`

> **Note:** This is for the built-in `format()` function, **not** the [string `.format()` method](https://www.codecademy.com/resources/docs/python/strings/format).

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo`` code snippet.
- An "Example" section with a ` ```py `code snippet.
- A “Codebyte Example” section with a ` ```codebyte/python ` snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by David-OC17**
### Description

This pull request is for the integration of a new entry for the Python built-in format() function. This is for the built-in format() function, not the string .format().

### Type of Change

- Adding a new entry

### Linked issue
Closes #2525

### CODE REVIEW

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Few changes from my side

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+format(value, format_specification)
+
+## Example
+In the following example we format a float in a variable called pi, which stores an aproximation of the value of pi. The format is done using specifications that denote that the output should be a string that is the provided float, but only using two decimal places.
+
+pi = 3.14159
+formatted = format(pi, '.2f')
+print(formatted)
+
```
Comment (dakshdeepHERE) [inline content/python/concepts/built-in-functions/terms/format/format.md:34]:
```suggestion

```python
pi = 3.14159
formatted = format(pi, '.2f')
print(formatted)
```

```

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Output:
+3.14
+
```
Comment (dakshdeepHERE) [inline content/python/concepts/built-in-functions/terms/format/format.md:37]:
```suggestion
Output for above code is:

```shell
3.14
```

```

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+## Codebyte Example (if applicable)
+
+The next example uses the **format()** function to convert an integer to a binary and print its string representation.
+
+value = 42
+formatted = format(value, 'b')  # Format value as binary
+print(formatted)
+
```
Comment (dakshdeepHERE) [inline content/python/concepts/built-in-functions/terms/format/format.md:45]:
```suggestion

```python
value = 42
formatted = format(value, 'b')  # Format value as binary
print(formatted)
```

```

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+Output:
+101010
```
Comment (dakshdeepHERE) [inline content/python/concepts/built-in-functions/terms/format/format.md:47]:
```suggestion
Output for the above code is:
```shell
101010
```

```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]

**Comment by dakshdeepHERE** [comment]
@David-OC17 Try to use tags with each code snippets. Please read [guide](https://www.codecademy.com/resources/docs/contribution-guide).

**Comment by David-OC17** [comment]
I just committed the changes requested by @dakshdeepHERE

**Comment by dakshdeepHERE** [comment]
> I just committed the changes requested by @dakshdeepHERE

Awsm tag the maintainers now for further reviews. :)

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+formatted = format(pi, '.2f')
+print(formatted)
+```
+
+Output for above code is:
+```shell
+3.14
+```
+
+## Codebyte Example (if applicable)
```
Comment (dakshdeepHERE) [inline content/python/concepts/built-in-functions/terms/format/format.md:38]:
```suggestion
## Codebyte Example
```

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+The next example uses the **format()** function to convert an integer to a binary and print its string representation.
+
+```python
```
Comment (dakshdeepHERE) [inline content/python/concepts/built-in-functions/terms/format/format.md:42]:
```suggestion
```codebyte/python
```

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+value = 42
+formatted = format(value, 'b')  # Format value as binary
+print(formatted)
+```
+
+Output for the above code is:
+```shell
+101010
+```
```
Comment (dakshdeepHERE) [inline content/python/concepts/built-in-functions/terms/format/format.md:51]:
```suggestion
```
no need to attach output section for codebyte

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+3.14
+```
+
+## Codebyte Example (if applicable)
+
+The next example uses the **format()** function to convert an integer to a binary and print its string representation.
+
+```python
+value = 42
+formatted = format(value, 'b')  # Format value as binary
```
Comment (dakshdeepHERE) [inline content/python/concepts/built-in-functions/terms/format/format.md:44]:
```suggestion
formatted = format(value, 'b')  
```
no comments in codebytes

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+CatalogContent:
+  - 'learn-git-&-github'
+---
+
+The **format()** function returns a string from an input value, formatted to the specified specifications.
+
+## Syntax
+
+format(value, format_specification)
+
```
Comment (dakshdeepHERE) [inline content/python/concepts/built-in-functions/terms/format/format.md:22]:
```suggestion

format(value, format_specification)

```

please write syntax snippet like this:
image

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Few more changes @David-OC17

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
+---
+Title: '.format()'
+Description: The format() function in Python is a built-in function that is used to format different types of objects into strings.
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/format/format.md:3]:
```suggestion
Description: 'Used to format different types of objects into strings.'
```

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+Subjects:
+  - 'A subject name'
+  - 'A second subject name'
+  - 'An nth subject name'
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/format/format.md:7]:
```suggestion
Subjects:
  - 'Computer Science'
```

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+Tags:
+  - 'Functions'
+  - 'Methods'
+  - 'Strings'
+  - 'built-in-funtions'
+CatalogContent:
+  - 'learn-git-&-github'
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/format/format.md:14]:
```suggestion
CatalogContent:
- 'learn-python-3'
- 'paths/computer-science'
```

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+---
+
+The **format()** function returns a string from an input value, formatted to the specified specifications.
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/format/format.md:17]:
```suggestion
The **format()** function returns a string from an input value, formatted to the provided specifications.
```

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+## Syntax
+
+```pseudo
+format(value, format_specification)
+```
+
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/format/format.md:24]:
```suggestion
Need a sentence explaining the syntax and an explanation of at least some format specifications.
```

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+## Codebyte Example
+
+The next example uses the **format()** function to convert an integer to a binary and print its string representation.
+
+```codebyte/python
+value = 42
+formatted = format(value, 'b')  
+print(formatted)
+```
+
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/format/format.md:49]:
```suggestion
```
Remove trailing blank line.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@David-OC17 In the interest of moving this along, I implemented the last of @dakshdeepHERE's suggestions and did a first maintainer review.

**Review by SSwiniarski — APPROVED** [review]
@David-OC17 I did the updates myself to move this along. I'm now sending it for a second review.

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+  - 'Functions'
+  - 'Methods'
+  - 'Strings'
+  - 'built-in-funtions'
+CatalogContent:
+- 'learn-python-3'
+- 'paths/computer-science'
+---
+
+The **format()** function returns a string from an input value, formatted to the provided specifications.
```
Comment (caupolicandiaz) [inline content/python/concepts/built-in-functions/terms/format/format.md:16]:
```suggestion
The **`format()`** function returns a string from an input value, formatted to the provided specifications.
```

File: content/python/concepts/built-in-functions/terms/format/format.md [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+format(value, format_specification)
+```
+
+Where `value` is the value to format, and `format_specification` is the format specification to use.
+
+Example format specifiers:
+
+| Specifier | Meaning                                |
+|:---------:|----------------------------------------|
+| b         | Binary format.                         |
```
Comment (caupolicandiaz) [inline content/python/concepts/built-in-functions/terms/format/format.md:30]:
```suggestion
| `b`         | Binary format                         |
```

**Comment by SSwiniarski** [comment]
@caupolicandiaz I went ahead and implemented your suggestions. Do we look good to merge?

**Review by caupolicandiaz — APPROVED** [review]
@David-OC17 @SSwiniarski LGTM

**Comment by David-OC17** [comment]
Looks good to merge. Just started merge of main-->this branch, in order to be up-to-date, and later merge the these changes back to main.

**Pull request merged**